src/ripple/net/impl/HTTPStream.cpp

```diff
@@ -0,0 +1,207 @@
+{
+    return std::make_unique<SSLStream>(config, strand, j);
+}
+
+boost::asio::ip::tcp::socket&
+SSLStream::getStream()
+{
+    return stream_->next_layer();
+}
+
+bool
+SSLStream::connect(
+    std::string& errorOut,
+    std::string const host,
+    std::string const port,
+    boost::asio::yield_context& yield)
+{
+    using namespace boost::asio;
+    using namespace boost::beast;
+
+    boost::system::error_code ec;
+
+    auto fail = [&errorOut](std::string const& errorIn) {
+        errorOut = errorIn;
+        return false;
+    };
+
+    ip::tcp::resolver resolver{strand_.context()};
+    auto const results = resolver.async_resolve(host, port, yield[ec]);
+    if (ec)
+        return fail("async_resolve");
+
+    try
+    {
+        stream_.emplace(strand_.context(), ssl_ctx_.context());
+    }
+    catch (std::exception const& e)
+    {
+        return fail(std::string("exception: ") + e.what());
+    }
+
+    ec = ssl_ctx_.preConnectVerify(*stream_, host);
+    if (ec)
+        return fail("preConnectVerify");
+
+    boost::asio::async_connect(
+        stream_->next_layer(), results.begin(), results.end(), yield[ec]);
+    if (ec)
+        return fail("async_connect");
+
+    ec = ssl_ctx_.postConnectVerify(*stream_, host);
+    if (ec)
+        return fail("postConnectVerify");
+
+    stream_->async_handshake(ssl::stream_base::client, yield[ec]);
+    if (ec)
+        return fail("async_handshake");
+
+    return true;
+}
+
+void
+SSLStream::asyncWrite(
+    request& req,
+    boost::asio::yield_context& yield,
+    boost::system::error_code& ec)
+{
+    boost::beast::http::async_write(*stream_, req, yield[ec]);
+}
+
+void
+SSLStream::asyncRead(
+    boost::beast::flat_buffer& buf,
+    parser& p,
+    bool readSome,
+    boost::asio::yield_context& yield,
+    boost::system::error_code& ec)
+{
+    if (readSome)
+        boost::beast::http::async_read_some(*stream_, buf, p, yield[ec]);
+    else
+        boost::beast::http::async_read(*stream_, buf, p, yield[ec]);
+}
+
+RawStream::RawStream(
+    Config const& config,
+    boost::asio::io_service::strand& strand,
+    beast::Journal j)
+    : strand_(strand)
+{
+}
+
+std::unique_ptr<HTTPStream>
+RawStream::getUnique(
+    Config const& config,
+    boost::asio::io_service::strand& strand,
+    beast::Journal j)
+{
+    return std::make_unique<RawStream>(config, strand, j);
+}
+
+boost::asio::ip::tcp::socket&
+RawStream::getStream()
+{
+    return *stream_;
+}
+
+bool
+RawStream::connect(
+    std::string& errorOut,
+    std::string const host,
+    std::string const port,
+    boost::asio::yield_context& yield)
+{
+    using namespace boost::asio;
+    using namespace boost::beast;
+
+    boost::system::error_code ec;
+
+    auto fail = [&errorOut](std::string const& errorIn) {
+        errorOut = errorIn;
+        return false;
+    };
+
+    ip::tcp::resolver resolver{strand_.context()};
+    auto const results = resolver.async_resolve(host, port, yield[ec]);
```

Comment:
As above, `results` is a very generic name, and since it's coming in the middle of the function, and isn't the result that's actually returned by the function, I think another name would be better. How about "resolved", "ips" (or something similarly descriptive), or "connectTargets".

Reply:
Done